protocol/messenger.go

```diff
@@ -2343,6 +2343,12 @@ func (m *Messenger) SendChatMessages(ctx context.Context, messages []*common.Mes
 	return &response, nil
 }
 
+func (m *Messenger) isOnlyMyselfJoinedChat(chat *Chat, rawMessage common.RawMessage) bool {
```

Comment:
mmh, not sure exactly, but I think the logic should be simpler, 
we should mark the message as `sent` only if 
1) Is a private group chat 
and 
2) The message is not sent to anyone (i.e Recipients are nil or 0)

If we send to someone (even ourselves, in case we have paired messages), then we should wait for the the dispatched event and so it should no be marked as `sent`.

So I think the logic should simply be:

```
  return chat.ChatType == ChatTypePrivateGroupChat && len(rawMessage.Recipients) == 0
```

Also I would name the function differently, something like `propagatedToTheNetwork`/`dispatchedOverTheNetwork`, i.e something that indicates that we want to know whether the message has been sent through waku or not.
Maybe @Samyoul has a good idea about naming it.

Reply:
Hmm, funny one, the func is positioned after the despatch but the question is not dependant on any altered state during the despatch. ie we can know before despatch if the message has a recipient, so asking a question past tense implies a change.

Maybe something like `isDispatchable()` would be better, as this naming is only a question of state rather than any actions performed.

Reply:
Also this leaves the function to be expanded and used in other contexts, rather than in this specific case.

Reply:
It's dependent on altered state in some sense though, if you look at `dispatch`, we set `Recipients` in case of `ChatTypePrivateGroupChat`.
We could re-calculate it as we have `chat` and the `MessageType`, but it's a bit expensive, so it would be probably better to piggy-back on the value that has been calculated in `dispatch`.
What do you think?

Reply:
Ah good point, I see that. Hmm, the only thing that I'd say is that this function is agnostic to whether a message has actually been despatched, it only cares if it could be despatched.

And alternative path could be to add a field on the `RawMessage` struct called `Despatched` which will be set by the despatcher `(p *MessageProcessor) sendPrivate` and can be queried at various points in the `Message`'s lifetime.

I suppose it depends on whether we want to know has the message **been** despatched or just if it **can be** despatched.

But if we add a `Despatched` field to the `RawMessage{}` struct, and set it to true only if `(p *MessageProcessor) sendPrivate` is successfully fired on the message, then we could do without this function and change the logic in `sendChatMessage()` to something like

```go
if chat.ChatType == ChatTypePrivateGroupChat && !rawMessage.Recipients.Despatched {
	message.OutgoingStatus = common.OutgoingStatusSent
}
```

Reply:
ah good idea,
actually `RawMessage` has already a `Sent` field, and we use that for re-transmission, so it makes sense to set that flag if it's not going out, so we don't even try to re-send it.
After we can check that field to set the `OutgoingStatus` i.e
```
if rawMessage.Sent {
  message.OutgoingStatus = common.OutgoingStatusSent
} else {
message.OutgoingStatus = common....Pending
}
```